source/recipe.rst

```diff
@@ -1,298 +1,225 @@
+   src_mtype_i src_region_i dst_mtype_i dst_region_i
+   =========== ============ =========== ============
+   3           -1           0           1
+   4           -1           0           1
+   5           -1           0           1
+   =========== ============ =========== ============
 
-- ``fromMType`` the `mtype` of the pre-synaptic cell
-- ``toMType`` the `mtype` of the post-synaptic cell
-- ``fromBranchType`` the classification of the pre-synaptic branch. May be one of
-  the following:
-  - ``*`` to match all branches
-  - ``soma`` to match the soma
-  - ``dendrite`` to match all dendrites
-  - ``basal`` for basal dendrites
-  - ``apical`` for apical dendrites
-- ``toBranchType`` the classification of the post-synaptic branch. May
-  also be referred to as ``type``, for backwards compatibility.
-
-ConnectionRules
-~~~~~~~~~~~~~~~
+Components
+----------
 
-These rules determine the distribution of synapses. They may take the
-following form::
+``bouton_distances``
+^^^^^^^^^^^^^^^^^^^^
 
-    <ConnectionRules>
-      <rule fromMType="L1_NGC-DA" toMType="*" bouton_reduction_factor= "0.114" active_fraction= "0.50" cv_syns_connection= "0.25" />
-      <rule fromMType="L1_HAC" toMType="L1_DAC" bouton_reduction_factor= "0.13" active_fraction= "0.50" cv_syns_connection= "0.25" />
-    </ConnectionRules>
+*Optional*, used by Functionalizer.
 
-.. note::
-   In older recipes, the rules take the form of:
-   ::
+Minimum distances for synapses, as measured along the branch length, starting at the soma.
+Synapses with less than the specified distance will be removed by Functionalizer.
 
-      <mTypeRule from="L1_HAC" to="L1_DAC" />
+.. table::
 
-   which will be translated into:
-   ::
+   =============================== =========== ===
+   Property                        Requirement Description
+   =============================== =========== ===
+   ``excitatory_synapse_distance`` Optional    The minimum distance from the soma for a synapse of post-synaptic excitatory cells. Defaults to 5µm.
+   ``inhibitory_synapse_distance`` Optional    The minimum distance from the soma for a synapse of post-synaptic inhibitory cells. Defaults to 25µm.
+   =============================== =========== ===
 
-      <rule fromMType="L1_HAC" toMType="L1_DAC" />
+``bouton_interval``
+^^^^^^^^^^^^^^^^^^^
 
-   automatically.
+*Required*, used by TouchDetector.
 
-Every rule can be used to select a subset of connections using attributes
-with the prefixes:
+Distances used when transforming touch regions into synapse candidates.
 
-- ``from`` for the pre-synaptic matching requirement
-- ``to`` for the post-synaptic matching requirement
+.. table::
 
-And the following stems:
+   =============================== =========== ===
+   Property                        Requirement Description
+   =============================== =========== ===
+   ``min_distance``                Mandatory   The minimum distance at which two synapse candidates are placed from each other.
```

Comment:
and what the unit is?

Reply:
µm. one could comment generically that all distances are in µm unless otherwise mentioned :thinking:

Reply:
Cool, I'd probably put that everywhere so it's hard to miss, and now one ever asks.